src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/SimpleTypeModelBinder.cs

```diff
@@ -65,6 +87,13 @@ public Task BindModelAsync(ModelBindingContext bindingContext)
             }
             catch (Exception exception)
             {
+                if (exception.InnerException != null)
+                {
+                    // TypeConverter throws System.Exception wrapping the FormatException,
+                    // so we capture the inner exception.
+                    exception = ExceptionDispatchInfo.Capture(exception.InnerException).SourceException;
```

Comment:
Not sure why we need that because TypeConverter throws System.Exception wrapping FormatException. (tried few samples through a separate Console application)